Release cycle

@danilobellini what do you think to publish releases more often? For instance there are several bug fixes in the current unreleased version. Perhaps could you publish a 0.9.1 with those changes?

> [...] only what we need is that @gorakhargosh register project on https://readthedocs.org [...]

That had already been done. If it hadn't, anyone would be able to do that. It's at https://watchdog.readthedocs.io/ and I've just created it again at https://python-watchdog.readthedocs.io/en/latest/

> [...] and provide us url and badge code [...]

You mean the config to be used here in GitHub to add a docs pipeline CI through a webhook? Anyway, I just asked the author to add me as a maintainer of the first link. The webhook config doesn't suffice as no collaborator is able to change the project hooks here in this GitHub repository, just the project owner.

> [...], the responsible course of action would be to transfer maintainership (most importantly, the ability to cut a Github and PyPI release) to someone who can, i.e. one of you two. [...]

I'm able to push to PyPI a new release (I did that on v0.9.0).

> I am very short in spare time, are you OK @danilobellini, can I process the next week?

Yes, and so do I! I'll have some time next tuesday, and uploading a new release to PyPI is quite easy and fast. Just tell me when to upload it to PyPI (or something else, if you need help).

> [...] I totally forgot the fact that it does not work on 3.8 [...]

Actually, v0.9.0 works on Python 3.8.0 (I've just tested here on Arch Linux using the [python38 3.8.0-1](https://aur.archlinux.org/packages/python38/) AUR package):

```
py38 run-test: commands[0] | py.test
============================= test session starts ==============================
platform linux -- Python 3.8.0, pytest-5.2.2, py-1.8.0, pluggy-0.13.0
cachedir: .tox/py38/.pytest_cache
rootdir: /home/danilo/code/watchdog, inifile: tox.ini
plugins: cov-2.8.1, timeout-1.3.3
collected 66 items                                                             

tests/test_delayed_queue.py .                                            [  1%]
tests/test_emitter.py .........s...                                      [ 21%]
tests/test_inotify_buffer.py ......                                      [ 30%]
tests/test_inotify_c.py .                                                [ 31%]
tests/test_observer.py .....                                             [ 39%]
tests/test_skip_repeats_queue.py ....                                    [ 45%]
tests/test_snapshot_diff.py .......                                      [ 56%]
tests/legacy/test_watchdog_events.py .....................               [ 87%]
tests/legacy/test_watchdog_observers_api.py ......                       [ 96%]
tests/legacy/test_watchdog_observers_polling.py ..                       [100%]

=============================== warnings summary ===============================
.tox/py38/lib/python3.8/site-packages/watchdog/utils/bricks.py:175
  /home/danilo/code/watchdog/.tox/py38/lib/python3.8/site-packages/watchdog/utils/bricks.py:175: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.9 it will stop working

[...]

================== 65 passed, 1 skipped, 1 warnings in 23.75s ==================
___________________________________ summary ____________________________________
  py38: commands succeeded
  congratulations :)
```

To run that, I just edited `tox.ini` to install `pytest-timeout` in the `py38` environment in the `v0.9.0` tag version. Is there any Python 3.8.0-specific bug I'm missing? If so, I think that should be a new issue to be fixed ASAP. Trying to run the the current master (a60664426498a67fecebb659bb8ab9d8f53640c4) gives an error because of a dangling thread:

```
==================================== ERRORS ====================================
______________________ ERROR at teardown of test___init__ ______________________

    @pytest.fixture(autouse=True)
    def no_thread_leaks():
        """
        Fail on thread leak.
        We do not use pytest-threadleak because it is not reliable.
        """
    
        yield
    
        if sys.version_info < (3,):
            return
    
        main = threading.main_thread()
>       assert not [th for th in threading._dangling
                    if th is not main and th.is_alive()]
E       assert not [<PollingEmitter(Thread-70, stopped daemon 140145821878016)>]

main       = <_MainThread(MainThread, started 140145857795904)>

tests/conftest.py:38: AssertionError
```

--- a/README.rst
+++ b/README.rst
@@ -134,7 +134,7 @@ Install from source:
     $ python -m pip install -e .
 
     # or to install the watchmedo utility:
-    $ python -m pip install -e .[watchmedo]
+    $ python -m pip install -e ".[watchmedo]"
 
 
 Installation Caveats

--- a/changelog.rst
+++ b/changelog.rst
@@ -6,7 +6,7 @@ Changelog
 0.10.0
 ~~~~~~
 
-unreleased • `full history <https://github.com/gorakhargosh/watchdog/compare/v0.9.0...master>`__
+2019-12-20 • `full history <https://github.com/gorakhargosh/watchdog/compare/v0.9.0...v0.10.0>`__
 
 Breaking Changes
 ================

--- a/src/watchdog/version.py
+++ b/src/watchdog/version.py
@@ -20,7 +20,7 @@
 # When updating this version number, please update the
 # ``docs/source/global.rst.inc`` file as well.
 VERSION_MAJOR = 0
-VERSION_MINOR = 9
+VERSION_MINOR = 10
 VERSION_BUILD = 0
 VERSION_INFO = (VERSION_MAJOR, VERSION_MINOR, VERSION_BUILD)
 VERSION_STRING = "%d.%d.%d" % VERSION_INFO

